android-auto-app/build.gradle.kts

```diff
@@ -35,7 +35,6 @@ android {
     if (buildFromSource.toBoolean()) {
       jniLibs.pickFirsts.add("**/libc++_shared.so")
     }
-    jniLibs.pickFirsts.add("**/libmapbox-common.so")
```

Comment:
The same here

Reply:
@Chaoba with previous gradle plugin it was more of an AGP bug that `so` from previous build was not cleared properly leading to duplicate error. Cleaning `build` directories fixed that. If that issue is fixed with new gradle plugin - we indeed do not need it here I guess.
@ank27 any more comments here?

Reply:
Yes correct, and since gl-native exclude `common.so`  as well, we don't need this here.

Reply:
the common.so could be duplicated when building from source, and it was 

```
if (buildFromSource.toBoolean()) {
    packagingOptions {
      pickFirst("**/libc++_shared.so")
      pickFirst("**/libmapbox-common.so")
    }
  }
}
```

before the [v10.4.0-beta.1 bump](https://github.com/mapbox/mapbox-maps-android/pull/1160/files#diff-8cff73265af19c059547b76aca8882cbaa3209291406f52df1dafbbc78e80c46L59-L60), we should probably move `jniLibs.pickFirsts.add("**/libmapbox-common.so")` to the `if (buildFromSource.toBoolean())` closure.

Reply:
@pengdev as I've described [here](https://github.com/mapbox/mapbox-maps-android/pull/1175#discussion_r812630460) those conditions were related to an issue in older AGP and could be fixed locally by cleaning previous build files. IMO we should remove it from end application gradle files to see if issue is resolved now and even if it's not - AFAIR we have a step in build from source section noting to call `make clean` before building from source. If we leave it - we may simply hide some error.

Reply:
Earlier releases of gl-native did not exclude `common.so` explicitly from `packagingOptions`, i guess that's why pickFirst was added in the above config. If i understand correctly even building from source, the libs files are packaged as per `packaginOptions` in gradle. since now we exclude `common.so` from package with 10.4.0-beta.2, it shouldn't matter if we `pickFirst` in the config above. But we can keep it just be sure, wdyt? 🤔

Reply:
we can remove it and if we encounter issues while building from source, we add it back 🙂

Reply:
> we can remove it and if we encounter issues while building from source, we add it back 🙂 Better to verify build from source first.

Attached internal ticket points to this pr which has `buildFromSource` passed. I think that would be fine to verify this behaviour?